src/navigation-plan.js

```diff
@@ -101,3 +107,13 @@ function hasDifferentParameterValues(prev, next) {
 
   return false;
 }
+
+function getInstructionBaseUrl(instruction) {
```

Comment:
The baseUrl we need is the concatenation of all ancestor instruction's baseUrls. Is this guaranteed to be correct?